### PR TITLE
MONIT-25613 Send logs via proxy foundations + UTs

### PR DESCRIPTION
**Pull request by amitw-vmware**
Foundational changes to send logs to WF via the Proxy + Unit Tests

See design doc for details: https://docs.google.com/document/d/1Y-cLQFwvt-bVBlhyWCZJRHD_BRSya03aQtcl2dh4TJ0/edit#

### CODE REVIEW

File: java-lib/src/main/java/com/wavefront/api/LogAPI.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    /**
+     * Ingest a batch of logs.
+     *
+     * @param logBatch batch of logs to be reported
+     * @return HTTP response
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("le-mans/v1/streams/ingestion-pipeline-stream")
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/api/LogAPI.java:24]: nit question: what does `le-mans` stand for in this path?

File: java-lib/src/main/java/com/wavefront/api/agent/AgentConfiguration.java [hunk 2]
```diff
@@ -287,6 +321,26 @@ public void setSpanLogsDisabled(boolean spanLogsDisabled) {
...
+
+  public void setLogsDisabled(boolean logsDisabled) {
+    this.logsDisabled = logsDisabled;
+  }
+
+  public String getLogServerEndpointUrl() {
+    return logServerEndpointUrl;
+  }
+
+  public void setLogServerEndpointUrl(String logServerEndpointUrl) { this.logServerEndpointUrl = logServerEndpointUrl; }
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/api/agent/AgentConfiguration.java:336]: nit: format

File: java-lib/src/main/java/com/wavefront/api/agent/AgentConfiguration.java [hunk 3]
```diff
@@ -287,6 +321,26 @@ public void setSpanLogsDisabled(boolean spanLogsDisabled) {
...
+
+  public String getLogServerToken() {
+    return logServerToken;
+  }
+
+  public void setLogServerToken(String logServerToken) { this.logServerToken = logServerToken; }
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/api/agent/AgentConfiguration.java:342]: nit: format

File: java-lib/src/main/java/com/wavefront/api/agent/AgentConfiguration.java (hunk 2 above)
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/api/agent/AgentConfiguration.java:336]: nit: format

File: java-lib/src/main/java/com/wavefront/api/agent/AgentConfiguration.java (hunk 3 above)
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/api/agent/AgentConfiguration.java:342]: nit: format

File: java-lib/src/main/java/com/wavefront/api/agent/ValidationConfiguration.java [hunk 4]
```diff
@@ -108,6 +132,16 @@ public int getSpanAnnotationsValueLengthLimit() {
     return spanAnnotationsValueLengthLimit;
   }
 
+  public int getLogAnnotationsKeyLengthLimit() { return logAnnotationsKeyLengthLimit; }
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/api/agent/ValidationConfiguration.java:135]: nit: format

File: java-lib/src/main/java/com/wavefront/ingester/ReportLogIngesterFormatter.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            Long timestamp = AbstractIngesterFormatter.getLogTimestamp(log.getAnnotations(), customLogTimestampTags);
+            log.setTimestamp(timestamp);
+            String message = AbstractIngesterFormatter.getLogMessage(log.getAnnotations(), customLogMessageTags);
+            log.setMessage(message);
+            return log;
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+            return null;
+        }
+
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/ReportLogIngesterFormatter.java:73]: nit: format spaces

**Review by joannak-vmware — APPROVED** [review]
Couple of nit comments

File: java-lib/src/main/java/com/wavefront/ingester/ReportLogIngesterFormatter.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                }
+            }
+            log.setHost(host);
+            Long timestamp = AbstractIngesterFormatter.getLogTimestamp(log.getAnnotations(), customLogTimestampTags);
+            log.setTimestamp(timestamp);
+            String message = AbstractIngesterFormatter.getLogMessage(log.getAnnotations(), customLogMessageTags);
+            log.setMessage(message);
+            return log;
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
```
Comment (laullon) [inline java-lib/src/main/java/com/wavefront/ingester/ReportLogIngesterFormatter.java:70]: log ?

**Review by laullon — APPROVED** [review]

**Pull request merged**